### DISCUSSION TITLE
Is it possible to compile into a static library for small embedded projects?

### DISCUSSION THREAD

**Discussion by andy27182** [eProsima/Fast-DDS#4004] [Q&A]
Hi
I am learning to use DDS. I can compile and use it simply on Linux according to the tutorial. The complete construction of DDS depends on many libs such as Asio, TinyXML2, OpenSSL. I hope to use DDS on some small MCU devices. These devices may have different operating systems, and Only C language support is required, so does Fast-DDS have build configurations to select modules, and build as a static library file.

**Comment by JLBuenoLopez** [comment]
@AndrewFrank-zxy 

It is possible to build Fast DDS statically. In order to remove OpenSSL dependency, you will need to disable the [TLS CMake option](https://fast-dds.docs.eprosima.com/en/latest/installation/configuration/cmake_options.html). It is also advisable to build Fast DDS with the THIRDPARTY CMake option set to FORCE so Asio and TinyXML are built within the Fast DDS static library, but these dependencies cannot be removed.

eProsima has options for embedded resources like:

* [Micro XRCE-DDS](https://eprosima.com/index.php/products-all/eprosima-micro-xrce-dds)
* [Safe DDS](https://eprosima.com/index.php/products-all/eprosima-safe-dds). This implementation is specially focused on [modularity](https://safe-dds.docs.eprosima.com/architecture/architecture.html#architecture).

**Reply by andy27182** [reply]
Thank you for your reply.
It seems that micro XRCE-DDS and Safe DDS are two best practices for Embedded System and I will continue to learn about them.

Based on what I know so far, it seems that micro XRCE-DDS is like a DDS adapter (with a client and a agent). If every node in a distributed system communicate through micro XRCE-DDS rather than standard dds middleware, Can this system work well?
